api/src/main/java/org/pmiops/workbench/cdr/CdrVersionContext.java

```diff
@@ -24,7 +28,23 @@ public static void clearCdrVersion() {
     cdrVersion.remove();
   }
 
+  @Nonnull
   public static DbCdrVersion getCdrVersion() {
-    return cdrVersion.get();
+    DbCdrVersion version = cdrVersion.get();
+    if (version == null) {
+      throw new ServerErrorException("No CDR version specified!");
+    }
+
+    return version;
+  }
+
+  /**
+   * BigQueryService.getBigQueryService() operates in two modes: with and without a CDR context
+   *
+   * @return the CDR Context's BigQuery project if there is a CDR in context, null if not
+   */
+  @Nullable
+  public static String nullableGetBigQueryProject() {
```

Comment:
nit: I think returning `Optional<String>` or the more generic `Optional<DbCdrVersion>` might be cleaner (if so - would just change the name to `maybe` instead of `nullable`)